app/scripts/modules/core/src/managed/constraints/ConstraintCard.tsx

```diff
@@ -22,26 +27,93 @@ const logUnsupportedConstraintError = (type: string) => {
   );
 };
 
+const overrideConstraintStatus = (
+  application: Application,
+  options: Omit<IUpdateConstraintStatusRequest, 'application'>,
+) =>
+  ManagedWriter.updateConstraintStatus({
+    application: application.name,
+    ...options,
+  }).then(() => {
+    const dataSource: ApplicationDataSource<IManagedApplicationEnvironmentSummary> = application.getDataSource(
+      'environments',
+    );
+
+    // Here we wait to say things are fully done until we attempt to refresh, but don't
+    // reject. If the refresh fails things are going to be in a bad state anyway,
+    // so we don't want to wrongly imply that the override didn't take effect
+    // just because we're unable check.
+    return dataSource.refresh().catch(() => null);
+  });
+
 export interface IConstraintCardProps {
+  application: Application;
+  environment: string;
+  version: string;
   constraint: IStatefulConstraint;
   className?: string;
 }
 
-export const ConstraintCard = ({ constraint, className }: IConstraintCardProps) => {
+export const ConstraintCard = ({ application, environment, version, constraint, className }: IConstraintCardProps) => {
   const { type, status } = constraint;
 
+  const [actionPending, setActionPending] = useState(false);
+  const [actionFailed, setActionFailed] = useState(false);
```

Comment:
It would be nice to have a single variable represent these two states. Something like the following

```js
const [actionStatus, setActionStatus] = useState<'not-initiated' | 'pending' | 'success' | 'error'>('not-initiated');
```

This would prevent accidentally setting an invalid state such as `actionPending=true` and `actionFailed=true`

If we need additional information for each state, such as error message for `error`, then we could extend it the following way and have typescript help us with refining.

```js
type ActionStatus = {
  status: 'not-initiated'
} | {
  status: 'pending'
} | {
  status: 'success'
} | {
  message: string,
  status: 'error'
};

```

Reply:
[done!](https://github.com/spinnaker/deck/pull/8124/files#diff-05e51702972fa4560063447334bc15f9R61)